orders/OrderFormNotion.ts

```diff
@@ -36,26 +38,34 @@ export default class OrderForm {
             });
 
             for (let i of orders.results) {
-                if (!("last_edited_time" in i)) {
+                if (!("last_edited_time" in i) || !isFullPage(i)) {
                     console.warn(`Order form checker: Received a partial response from Notion, skipping.`);
                     continue;
                 }
 
                 const pageLastEditedTime = new Date(i.last_edited_time).getTime();
-
                 // Send a Discord message only after the first sync,
                 // and only if there is a new order form entry or an order form entry has changed.
                 if (!isFirstSync && (!this.idTimesMap[i.id] || pageLastEditedTime != this.idTimesMap[i.id])) {
-                    if (i.properties["Name"]?.type !== "title") {
-                        console.warn(`Order form checker: Name property is not properly formatted.`);
+                                    // given the required title, product name, and subtotal are not filled out, skip until next sync
+                    if (!this.completedSushiOrder(i)) {
+                        console.warn(`Order form checker: Received a partially filled out page ${i.id}, skipping until required title, product name, and subtotal are filled out`);
                         continue;
                     }
 
-                    try {
-                        const nameFromNotion = i.properties["Name"].title[0]?.plain_text ?? "";
-                        const rosterEntry = await this.notion.getRosterEntryFromName(nameFromNotion);
+                    if (i.properties["Submitter"]?.type != "people") {
+                        console.warn(`Order form checker: Submitter not "people" type`);
+                        continue;
+                    }
 
-                        this.bot.updateUsers(rosterEntry?.discordTag ?? null, i.properties);
+                    try {
+                        const submitterName = i.properties["Submitter"].people.at(0);
+                        if (submitterName != null && isFullUser(submitterName) && submitterName.name != null) {
+                            const rosterEntry = await this.notion.getRosterEntryFromName(submitterName.name);
+                            this.bot.updateUsers(rosterEntry?.discordTag ?? null, i);
+                        } else {
+                            throw Error("Submitter object malformed")
```

Comment:
nit: style (don't think the `new` does anything major for `Error`)

```suggestion
                            throw new Error("Submitter object malformed");
```